src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/Cmdlets/UpdatePackageCommand.cs

```diff
@@ -182,9 +195,23 @@ private async Task UpdateOrReinstallAllPackagesAsync()
                         new GatherCache(),
                         sourceCacheContext);
 
+                    // PackageReference projects don't support `Update-Package -Reinstall`. 
+                    var relevantProjects = Projects;
+                    if (Reinstall.IsPresent)
+                    {
+                        var groupedProjects = Projects.GroupBy(e => e is BuildIntegratedNuGetProject);
```

Comment:
Previously I received feedback about below is better for performance, maybe you can check.

https://github.com/NuGet/NuGet.Client/blob/77e1bdb266093fd1eafff90ef21ad55a2a448889/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L2055-L2062